### PR TITLE
feat: entropy stages initial design

### PR DESCRIPTION
**Pull request by Fildrance**

### CODE REVIEW

**Comment by Fildrance** [comment]
Possibly add anomaly interactions for lowering or slowing down entropy. 
Server 'eating gas' is wierd.
Cooling gases is very easy as it is now. It might be just not mattering much, as cooling down to zero might be question of 2-5 minutes (with RPD especially).
Need to add exact numbers for tables... need to add formula for tech cost

**Comment by Kadeo64** [comment]
So you make research more expensive, and then the only way to make it less expensive is to rely on salvage, which is a gamble in and of itself, or to rely on engi, who are most likely going to be dealing with something else, and probably already dislike science as it's a power intensive department already, with the added 'bonus' of research taking more time?

and the only benefit is 2 tier 3 disciplines.. (which, like, speed boots + arsenal might be useful during war ops, but tier 3s can't really combo all that much.)

cooling gas is an interesting idea but IMO it should not be a direct intake system, more like based on the temperature of the air around it, since most server rooms are already equipped with a freezer setup. To prevent just spacing server room make it shut off at low pressures, easy.

**Comment by Fildrance** [comment]
> So you make research more expensive, and then the only way to make it less expensive is to rely on salvage, which is a gamble in and of itself, or to rely on engi, who are most likely going to be dealing with something else, and probably already dislike science as it's a power intensive department already, with the added 'bonus' of research taking more time?
> 
> and the only benefit is 2 tier 3 disciplines.. (which, like, speed boots + arsenal might be useful during war ops, but tier 3s can't really combo all that much.) (edit: No, microreactor cells + svalinn is not a very good combo. high caps are better for svalinn, and if you want an auto-charging laser pistol go ask sec for a ALP you already have t3 arsenal if you have a svalinn)
> 
> Cooling gas is an interesting idea but IMO it should _not_ be a direct intake system, more like based on the temperature of the air around it (like real server rooms, but exaggerated since real server rooms are never freezing cold.), since most server rooms are already equipped with a freezer setup. To prevent just spacing server room make it shut off at low pressures, easy.
> 
> Also, most rounds last like, an hour. Science doesn't really get a lot of time to mess with tier 3s already unless they specifically rush one, and even then you're at salvage's mercy to collect the resources you need for it. This just exacerbates those problems, I think.

I added some stuff. I have a lot of 4+hours shifts, night shifts, so maybe my perspective is pretty specific. Would like to hear feedback, i guess.
Cooling gas inside server room is kinda nice idea, i really like it too, but not all servers going to have server ROOM, some of them have server inside RD room. Dunno. Thing to think about. Also i am not 100% sure this cooling idea is good coz currently atmos can create -273 C degree gas blindfolded in 3 minutes, its TOO easy and can be done using ZERO power, so i'm not sure its going to be THAT engaging mechanic. But oh well.

**Comment by Kadeo64** [comment]
> I added some stuff. I have a lot of 4+hours shifts, night shifts, so maybe my perspective is pretty specific. Would like to hear feedback, i guess. 

I would say about 70%+ of the game's population plays on the highpop LRP servers during peak hours. Most of these rounds last between 50 and 90 minutes. balancing a major mechanic around 4 hour rounds just shouldn't be done I think.

>Cooling gas inside server room is kinda nice idea, i really like it too, but not all servers going to have server ROOM, some of them have server inside RD room. Dunno. Thing to think about. 

mapping issue

>also i am not 100% sure this cooling idea is good coz currently atmos can create -273 C degree gas blindfolded in 3 minutes, its TOO easy and can be done using ZERO power, so i'm not sure its going to be THAT engaging mechanic. But oh well.

rewards good atmos players for being good at their jobs

**Comment by Fildrance** [comment]
> rewards good atmos players for being good at their jobs

atmos players that i know consider such tasks as boring and not rewarding, as the only thing left to do after all job is done is to get drunk at the bar.

>mapping issue

It still is an issue, and not simple one. Silly station will require a buttload of time to fix such problem as there is no extra space to begin with. 

> balancing a major mechanic around 4 hour rounds just shouldn't be done I think
Its not balanced around, as much as it just blocks 'rushing' stuff. But as i said - i might be having specific perspective.

**Comment by ArtisticRoomba** [comment]
So I have a lot of thoughts about this, in no particular order:

I agree with the concept of making science have a time component and not just "spend points to instantly get (x, y)." I think it'll make things more fun and have research _feel like it's a big thing_ if it takes time. "The nukies are banging at our door, security, buy us more time so we can make better guns!" that sort of thing.

The server cooling problem. I think it's a good idea to get smart atmos techs involved, but frequently my atmos techs never get any cool tools that help them do their job, so they wouldn't really be _inclined_ to help them outside of roleplay.

Science is in this weird position right now where they can both research _and_ print the technology. This often leads to science hogging all of the technology themselves, burning all of the materials to print gear for themselves, and throwing their hands up in the air when someone asks for better gear. It would be way better for science to research technology that can be printed at the respective departmental lathe. That way science can ask for someone's skills or tools at their department, which helps with interdepartmental collaboration.

For cooling the server, Atmos can set up a small cooling loop with a freezer. It's basically enough to do the job. It's important to note that in real life, cooling a computer by putting it in a vacuum doesn't work out well. A radiator designed for convection with the air does not work well in space; there is no convection in space! So in reality spacing a room would overheat a server more than anything, there is nowhere for the energy to go. Another thing thing to note, with my whole station air circulation doc, this will give Atmos something to do, as this directly heats the air (#319).

As for the whole Anti-Entropy engine, be careful, as putting a load of 100kW on the grid is a pretty sizeable dent. For a Singulo, it'll lead to faster collector drain. It doesn't matter for a Tesla, and it gives Atmos something to do for a TEG (ensure their TEG is throttled properly). I'd also rather have this be an MV connection so Engineering can see what the hell is drawing 100kW. Of course, if Engineering doesn't have enough power, Science will just brown out, and the other departments around it as well. And then everyone will get mad at Science.

I like frezon being used as a valuable gas outside of selling it, but once again, it would be better if Atmos could print their relative technologies at their lathes, otherwise Atmos would see little to no benefit, and they would not be incentivized to make frezon. They would be spending their time making a gas helping science research fancy tools that they will likely keep to themselves.

You mention rerolling research randomness, but you also mention the idea of a research queue, which partially conflict. I don't mind the implementation of a research tree but you'd have to find ways to make it interesting.

The whole idea of introducing entropy to encourage collaboration between departments is cool, but right now this is kind of just forcing Engineering into this loop with not much direct benefit on Engineering's side. You're introducing power and atmos work to engineering, and you're forcing Science to beg engineering to assist them with it (as science's work slows to a crawl without the help), and all this time engineering can still have technology withheld from them if science is being genuinely greedy. I like the idea, but there are still some things that need tweaking.

**Comment by cohanna** [comment]
I think the dramatic point increase based on # of tech researched would "cut" a lot of content from what is pretty easily the most content lacking department. Point generation is already pretty dull and increasing the amount of points needed to research something would make the overall flow of the department even *more* slow, unengaging, and passive. Overall just a lot of sitting around waiting for both points to accumulate and other departments to *maybe* help out (less fun time with toys essentially). This kind of system would be better implemented after the various sub-departments of sci have been implemented/fleshed out to allow sci to both have more things to do and different avenues to generate these points.

**Comment by Fildrance** [comment]
Lack of useful stuff that sci department can provide to station is a very big problem, but this PR is not aiming to fix that. This concept is about 'polishing' one of basic mechanics of department. If it will be considered that all those mechanics not gonna be working until department interactions with sci is fixed - well, this design doc will need to wait for... a lot of time, sadly.

In my personal opinion lack of interest of other departments in interaction with sci is mostly due to amount of stuff that sci can create is mapped on start already (tons of RCD charges, holofan projectors, scrubbers) and those rarely end up all used or broken.

**Comment by Fildrance** [comment]
> Overall just a lot of sitting around waiting for both points to accumulate and other departments to _maybe_ help out (less fun time with toys essentially). 

Well, it motivates interaction - so you can either go and ask other departments to help in exchange for something, or you can try and do their work for yourself - find pacmans, terrorize station by draining all power and force some interactions this way...  its not that simple. There are options. Its just that you need to actually think how to spend your time effectively, and keep more things in mind.

**Comment by kurokoTurbo** [comment]
> > So you make research more expensive, and then the only way to make it less expensive is to rely on salvage, which is a gamble in and of itself, or to rely on engi, who are most likely going to be dealing with something else, and probably already dislike science as it's a power intensive department already, with the added 'bonus' of research taking more time?
> > and the only benefit is 2 tier 3 disciplines.. (which, like, speed boots + arsenal might be useful during war ops, but tier 3s can't really combo all that much.) (edit: No, microreactor cells + svalinn is not a very good combo. high caps are better for svalinn, and if you want an auto-charging laser pistol go ask sec for a ALP you already have t3 arsenal if you have a svalinn)
> > Cooling gas is an interesting idea but IMO it should _not_ be a direct intake system, more like based on the temperature of the air around it (like real server rooms, but exaggerated since real server rooms are never freezing cold.), since most server rooms are already equipped with a freezer setup. To prevent just spacing server room make it shut off at low pressures, easy.
> > Also, most rounds last like, an hour. Science doesn't really get a lot of time to mess with tier 3s already unless they specifically rush one, and even then you're at salvage's mercy to collect the resources you need for it. This just exacerbates those problems, I think.
> 
> I added some stuff. I have a lot of 4+hours shifts, night shifts, so maybe my perspective is pretty specific. Would like to hear feedback, i guess. Cooling gas inside server room is kinda nice idea, i really like it too, but not all servers going to have server ROOM, some of them have server inside RD room. Dunno. Thing to think about. Also i am not 100% sure this cooling idea is good coz currently atmos can create -273 C degree gas blindfolded in 3 minutes, its TOO easy and can be done using ZERO power, so i'm not sure its going to be THAT engaging mechanic. But oh well.

How do you create -273C gas? The limit in the game is 2.7K which is -270C

**Comment by Fildrance** [comment]
> How do you create -273C gas? The limit in the game is 2.7K which is -270C

Good point! it should be min value indeed. 

I am also currently considering changing server using gas to 'body temperature', as it will make stuff at least kinda dangerous and challenging.

File: src/en/space-station-14/departments/science/proposals/entropy.md [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+     line [30300,30800,31475,32225,33075,34025,35025,36100,37250,38450,39700,41000,42350,43750,45200,46700,48250,49825,51425,53100,54775,56500,58275,60075,61925,63775,65675,67625,69575,71575,73600,75650,77725,79850,82000,84150,86350,88575,90825,93100,95400,97725,100075,102450,104850,107275,109725,112200,114700,117200,119750,122325,124900,127500,130125,132800,135450,138150,140875,143600]
+
+```
+same numbers but in table:
+
+#### with 5000 cost tech
+
+| technologies researched count | next research cost |
+| --- | --- |
+|1 |5050 |
```
Comment (Fildrance) [inline src/en/space-station-14/departments/science/proposals/entropy.md:53]: hide big tables into spoilers!

**Comment by Fildrance** [comment]
From some more from discord discussions:
It seems like queue system and entropy is kinda nice and interesting, but choices research presents to players (which entropy/queue works as supplementary system to) are not - a lot of them are really rarely useful. Some - due to existance of pre-mapped item (sec stuff is not THAT great compared to provided by roundstart, most of atmos/engi stuff), some are just not 'useful' -  cooking and social stuff is very rarely even requested and have literally zero impact on round.
Even rerolling could not fix that problem fully - we have THAT much of questionably useful tech.

And the fact that we will rise cost of all that not useful tech is going to feel plain bad. So it might be good idea to conserve this concept for future (conceptual) in which we consider more then 80% of research technologies useful and having impact on round or at least **very** desirable... still inconclusive tho!

**Comment by Fildrance** [comment]
Following major discord discussion - its better to bind Entropy idea to https://github.com/space-wizards/docs/pull/336, and to implement / make changes publicly available befor Entropy-related features. And as researchable items count will go much lower - numbers on cost formula and graph should be changed to maintain core meaning. Entropy should not affect license costs.

**Review by Princess-Cheeseballs — CHANGES_REQUESTED** [review]
Is this still being worked on or has it been obsoleted by the science design doc?

If not this doc needs to be cleaned up a lot. The massive very specific tables make it very hard to parse and ideally we don't want design docs to be that specific as to have exact numerical balancing. These implementation details should be removed and in there stead should be a couple more statements about the design goals and reasoning why this is a good mechanic.

Overall, this does seem like a very good idea, hope to see it in the future.
